library/src/main/java/com/getstream/sdk/chat/notifications/options/StreamNotificationOptions.java

```diff
@@ -0,0 +1,150 @@
+package com.getstream.sdk.chat.notifications.options;
+
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.BitmapFactory;
+import android.os.Build;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+import androidx.core.app.NotificationCompat;
+
+import com.getstream.sdk.chat.R;
+import com.getstream.sdk.chat.model.Event;
+import com.google.firebase.messaging.RemoteMessage;
+
+/*
+ * Created by Anton Bevza on 2019-11-15.
+ */
+public class StreamNotificationOptions implements NotificationOptions {
+
+    private static final int DEFAULT_REQUEST_CODE = 999;
+
+    private String notificationChannelId;
+    private String notificationChannelName;
+    private NotificationChannel notificationChannel;
+    private NotificationCompat.Builder notificationBuilder;
+    private Intent defaultLauncherIntent;
+    private NotificationIntentProvider notificationIntentProvider;
+
+    @RequiresApi(api = Build.VERSION_CODES.O)
+    @Override
+    public NotificationChannel getNotificationChannel(Context context) {
+        if (notificationChannel == null) {
+            NotificationChannel channel = new NotificationChannel(
+                    getNotificationChannelId(context),
+                    getNotificationChannelName(context),
+                    NotificationManager.IMPORTANCE_DEFAULT
+            );
+            channel.setShowBadge(true);
+            return channel;
+        } else {
+            return notificationChannel;
+        }
+    }
+
+    @Override
+    public String getNotificationChannelId(Context context) {
+        if (notificationChannelId == null) {
+            return context.getString(R.string.default_notification_channel_id);
+        } else {
+            return notificationChannelId;
+        }
+    }
+
+    @Override
+    public String getNotificationChannelName(Context context) {
+        if (notificationChannelName == null) {
+            return context.getString(R.string.default_notification_channel_name);
+        } else {
+            return notificationChannelName;
+        }
+    }
+
+    @Override
+    public NotificationCompat.Builder getNotificationBuilder(Context context) {
+        if (notificationBuilder == null) {
+            return new NotificationCompat.Builder(context,
+                    getNotificationChannelId(context))
+                    .setAutoCancel(true)
+                    //TODO need default icons for notifications
+                    .setSmallIcon(R.mipmap.icon)
```

Comment:
There should be way to replace `R.mipmap.icon` with custom one. That's the first request which will come when we release notification manager. So `NotificationOptions` should have setter for icons:
```kotlin
interface NotificationOptions {

  fun setSmallIcon(int iconRes)
  fun setLargeIcon(int iconRes)

} 
```
And currently `R.mipmap.icon` is a stream logo. It shouldn't be there, if customer forgets to replace icon users will see some unknown logo and it will be extremely confusing. As icon is required by Android SDK I would at least replace `R.mipmap.icon` with just a white square.

Also:
- `R.mipmap.icon` should be renamed into `R.mipmap.stream_notification_icon`.
- Documentation should be updated about icon
- Warning should be added to logs with message like `Stream notification icon is not defined, default one is used!`